src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => (
+    isNaN(century)
+      ? person.sex === 'm'
+      : person.sex === 'm'
+      && Math.ceil(person.died / 100) === century
+  ));
+
+  const calculateSumMenAge = men.reduce((sum, { died, born }) => (
+    sum + died - born
+  ), 0);
```

Comment:
DRY (Dont Repeat Yourself) - you can move this logic to separate function and reuse it